tuna/rocmlir/rocmlir_worker.py

```diff
@@ -0,0 +1,165 @@
+
+    self.logger.info('Inserting results for job_id=%s', self.job.id)
+    query = gen_insert_query(obj, self.result_attr,
+                             self.dbt.results.__tablename__)
+    session.execute(query)
+    session.commit()
+    return True
+
+  def process_result(self, result_str: str):
+    """process tuning-run results"""
+    status = []
+    with DbSession() as session:
+
+      def actuator(func, result_str):
+        return func(session, result_str)
+
+      #retry returns false on failure, callback return on success
+      ret = session_retry(session, self.update_result_table,
+                          functools.partial(actuator, result_str=result_str),
+                          self.logger)
+      if not ret:
+        self.logger.warning('RocMLIR:  Unable to update database')
+
+    return status
+
+  def output_filename(self):
+    """Canonical name for tuningRunner.py output for a job."""
+    return f"tuning-results-{self.job.id}.tsv"
+
+  def step(self):
+    """Main functionality of the worker class. It picks up jobs in new state and executes them"""
+
+    if not self.get_job("new", "running", False):
+      #Sleep in case of DB contention
+      sleep(random.randint(1, 10))
+      return False
+
+    self.logger.info('Acquired new job: job_id=%s', self.job.id)
+    self.set_job_state('running')
+
+    try:
+      retcode,cmd_output = self.run_cmd()
+    except ValueError as verr:
+      self.logger.info(verr)
+      self.set_job_state('errored', result=verr)
+    else:
+      if retcode != 0:
+        msg = f"Error code {retcode}, output {cmd_output}"
+        self.logger.info(msg)
+        self.set_job_state('errored', result=msg)
+      else:
+        try:
+          with open(self.output_filename(), 'r', encoding='utf8') as results:
+            # https://stackoverflow.com/questions/49902843/avoid-parameter-binding-when-executing-query-with-sqlalchemy
+            string = results.read().replace(':', r'\:')
+            self.set_job_state('completed', result=string)
+            self.process_result(string)
+          os.remove(self.output_filename())
+        except OSError as exc:
+          # Most often FileNotFoundError.
+          self.logger.warning('Exception occurred while saving results of job %s:  %s',
+                              self.job.id, exc)
+          self.set_job_state('errored', result=repr(exc))
+
+    return True
+
+  def run_cmd(self):
+    """Run the actual workload"""
+    env_str = " ".join(self.envmt)
+    with DbSession() as session:
+      cft = self.dbt.config_table
+      query = session.query(cft).filter(cft.id == self.job.config)
+      config = query.all()
+      if len(config) > 1:
+        raise ValueError(f"More than one config matching ID {self.job.config}")
+      config_string = config[0].config_string()
+    cmd = env_str + f" python3 ./bin/tuningRunner.py --operation conv \
+                     --config='{config_string}' --mlir-build-dir `pwd` \
```

Comment:
--config is how to send tuningRunner.py just one config, instead of a file of them.
--tflops is an addition to include the winning TFlops value so I can record it.
--rocmlir_gen_flags lets me specify the GPU, which I'm only half-certain works correctly.

Reply:
The HIP_VISIBLE_DEVICES environment variable has worked in my experience.